src/components/GradeLevelFilter.jsx

```diff
@@ -63,20 +56,23 @@ export default function GradeLevelFilter() {
         transformOrigin={{ vertical: 'top', horizontal: 'left' }}>
         <FormControl component="fieldset" className={classes.popover}>
           <FormGroup>
-            {Array.from(gradeItems).map((grade) => (
-              <FormControlLabel
-                key={grade}
-                control={
-                  <Checkbox
-                    color="primary"
-                    checked={selectedGrades.has(grade)}
-                    onChange={() => toggleSelection(grade)}
-                    name="grade"
-                  />
-                }
-                label={grade}
-              />
-            ))}
+            {Object.entries(GradeLevel.values()).map(([val, stringRep]) => {
+              val = parseInt(val);
+              return (
+                <FormControlLabel
+                  key={val}
+                  control={
+                    <Checkbox
+                      color="primary"
+                      checked={grades.indexOf(val) > -1}
```

Comment:
I suggest leaving `grades` as a Set and setting its PropType to be a set. You'll have to convert an array to a set in the caller.